packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/component_builder.rs

```diff
@@ -48,17 +48,32 @@ impl ComponentBuilder {
                 NormalizedNode::Element(elm) => elm,
                 _ => unreachable!(),
             };
-            let extending = elm.extending.clone().map(|e| e.take_resolution());
-            if extending.is_none() {
+            if elm.extending.is_none() {
                 continue;
             }
-            let ref_resolution = extending.unwrap();
+            let ref_source = elm.extending.clone().unwrap();
             let component = &builder.components[idx];
-            let referent = &builder.components[ref_resolution.node_idx];
+            let referent = &builder.components[ref_source.idx()];
 
-            match Self::determine_extending(&ref_resolution, component, referent) {
+            match Self::determine_extending(ref_source, component, referent) {
                 Ok(extending) => {
-                    builder.components[idx].set_extending(extending);
+                    let extending_component = matches!(extending, Extending::Component(..));
+                    let component_type_changed =
+                        component.get_component_type() != referent.get_component_type();
+                    builder.components[idx].set_extending(Some(extending));
+
+                    if extending_component && component_type_changed {
+                        // Since we are extending a component where the component type changed,
+                        // the reference was inside an extend attribute and
+                        // `Expander::consume_extend_attributes` added an extra child
+                        // in case the referent was marked `extend_via_default_prop`
+                        // which would created an `Extending::Prop`.
+                        // The fact that we still have an `Extending::Component`
+                        // means that this extra child was not needed, so we delete it.
+                        let mut children = builder.components[idx].take_children();
+                        children.remove(0);
+                        builder.components[idx].set_children(children);
```

Comment:
I'm not thrilled with adding children to then be conditionally removed later. We'll discuss if there is another way.

Reply:
We could look up to see if the referent has `extend_via_default_prop` back when we attach the file in `Expander::consume_extend_attributes`.